authors/apps/articles/tests/test_article_api_endpoints.py

```diff
@@ -186,22 +181,44 @@ def test_delete_an_article_when_user_is_authorized(self):
         self.assertDictEqual(expected_dict, response.data)
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
-    def create_article_and_authenticate_test_user(self):
-        """
-        This method create an article and force authenticates a user
-        """
-        user = self.activated_user()
-        self.client.force_authenticate(user=user)
-        Article.objects.create(
-            title='whale',
-            description='fish',
-            body='In water',
-            author=Profile.objects.get(user=user.id))
+    def test_user_can_like_an_article(self):
+        user, article = self.create_article_and_authenticate_test_user()
+        self.assertFalse(article.is_liked_by(user))
+        response = self.client.post(self.like_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(article.is_liked_by(user))
 
-    def create_article_and_authenticate_test_user_2(self):
-        """
-        This method create an article and force authenticates a different user
-        """
-        self.create_article_and_authenticate_test_user()
-        user2 = self.create_another_user_in_db()
-        self.client.force_authenticate(user=user2)
+    def test_user_can_dislike_an_article(self):
+        user, article = self.create_article_and_authenticate_test_user()
+        self.assertFalse(article.is_disliked_by(user))
+        response = self.client.post(self.dislike_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertFalse(article.is_liked_by(user))
+
+    def test_user_can_get_the_like_status_for_an_article(self):
+        user, article = self.create_article_and_authenticate_test_user()
+        response = self.client.get(self.is_liked_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["is_liked"], False)
+        article.liked_by.add(user)
+        response = self.client.get(self.is_liked_article_url(article.slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["is_liked"], True)
+
+    def test_user_can_get_the_dislike_status_for_an_article(self):
```

Comment:
Thanks for the great work @Karuhanga. I am going to ask you for two changes 
1. I am kindly asking you to please add comments to your methods and classes.
2. And make your tests more modular, for example, this test is testing two functionalities at once. this could be further split up.
Otherwise, this is really great work.

Reply:
Thanks @FahdJamy .
I will be adding the comments shortly.
The second point though, I am actually testing for one thing here, except that I am doing it twice.
Do you think it needs a farther breakdown? Perhaps the comments would help with this as well?